pkg/controller/crdmirroring/crdmirroring_handlers.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crdmirroring
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog"
+
+	core "github.com/vmware-tanzu/antrea/pkg/apis/core/v1alpha2"
+	ops "github.com/vmware-tanzu/antrea/pkg/apis/ops/v1alpha1"
+	security "github.com/vmware-tanzu/antrea/pkg/apis/security/v1alpha1"
+	"github.com/vmware-tanzu/antrea/pkg/controller/crdmirroring/types"
+	legacycore "github.com/vmware-tanzu/antrea/pkg/legacyapis/core/v1alpha2"
+	legacyops "github.com/vmware-tanzu/antrea/pkg/legacyapis/ops/v1alpha1"
+	legacysecurity "github.com/vmware-tanzu/antrea/pkg/legacyapis/security/v1alpha1"
+)
+
+func (c *Controller) onNewCRDAdd(obj interface{}) {
+	var CRD metav1.Object
+	switch c.CRDName {
+	case NetworkPolicy:
+		CRD = obj.(*security.NetworkPolicy)
+	case ClusterNetworkPolicy:
+		CRD = obj.(*security.ClusterNetworkPolicy)
+	case Tier:
+		CRD = obj.(*security.Tier)
+	case ClusterGroup:
+		CRD = obj.(*core.ClusterGroup)
+	case ExternalEntity:
+		CRD = obj.(*core.ExternalEntity)
+	case Traceflow:
+		CRD = obj.(*ops.Traceflow)
```

Comment:
it should be converted to `metav1.Object` directly, no reason to be converted to concrete types first.

Reply:
Got it.